src/runtime/storage/firebase-storage.ts

```diff
@@ -66,6 +66,15 @@ class FirebaseKey extends KeyBase {
     }
   }
 
+  base(): string {
+    const str = this.toString();
+    return str.substring(0, str.length - this.arckey().length);
+  }
+
+  arckey(): string {
+    return this.location.substring(this.location.lastIndexOf('/') + 1);
```

Comment:
What exactly is `arckey` intended to be? Is it just matching the arc id? Are we sure that it won't contain `/` chars?

Reply:
renamed to arcId to be consistent with arcId in VolatileKey.
The FirebaseKey doesn't have a separate arcId (or how shell calls it arckey) field, and it is constructed as `${storage}/${arcid}` - where storage is `firebase://arcs-storage.firebaseio.com/AIzaSyBme42moeI-2k8WgXh-6YK_wYyjEXo4Oz8/${version}`, and version ends up being prefix of firebase key's `location`. 
i agree that relying on arcId not containing a '/' is probably suboptimal, but I'm not sure how else to get a hold of it, any recommendations welcome.